src/main/java/com/example/moviemanager/service/MovieService.java

```diff
@@ -1,31 +1,59 @@
 package com.example.moviemanager.service;
 
-import com.example.moviemanager.service.exception.MovieNotFoundException;
-import com.example.moviemanager.repository.model.Movie;
+import com.example.moviemanager.constantValues.MovieType;
 import com.example.moviemanager.repository.MovieRepository;
+import com.example.moviemanager.repository.model.Movie;
+import com.example.moviemanager.service.exception.FailedToRetrieveDataFromDatabaseException;
+import com.example.moviemanager.service.exception.MovieNotFoundException;
+import com.example.moviemanager.service.exception.UniqueMovieNotFoundException;
+import com.example.moviemanager.service.model.CreateMovieInfo;
+import com.example.moviemanager.service.model.ExistingMovieDetails;
+import com.example.moviemanager.service.model.UpdateMovieInfo;
+import com.example.moviemanager.service.model.api.ApiResponse;
+import com.example.moviemanager.service.model.api.MovieDetailsFromApi;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.dao.EmptyResultDataAccessException;
 import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
 
 import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
 @Service
 public class MovieService {
 
+    private static final String TV_SHOWS_URI =
+            "https://api.themoviedb.org/3/search/tv?api_key=77938ea875914a4c8ded6ccb9d840d27&query=";
```

Comment:
1. api key should not be hardcoded. It is sensitive information so, you should not expose it anywhere. And it should be configurable through application properties, but should not be added there. 
2. The base URI (https://api.themoviedb.org/) can be configured directly on RestTemplate if you would not create each instance here with RestTemplate() but instead would create a Bean which can be injected. The bean itself can be created with RestTemplateBuilder class. What is more base url should also be configurable through application properties.

Reply:
1. Removed API KEY from application properties and updated readme, so that in order to use app, you have to specify api key.
2. Created bean.